packages/web-sockets/src/websocket.ts

```diff
@@ -76,13 +85,73 @@ export class WebSocket extends InputGatedEventTarget<WebSocketEventMap> {
   static readonly READY_STATE_CLOSING = 2;
   static readonly READY_STATE_CLOSED = 3;
 
+  // Whether this instance was constructed by a user using the standard
+  // `new WebSocket()` constructor
+  readonly #userConstructed;
+
   #dispatchQueue?: ValueOf<WebSocketEventMap>[] = [];
   [kPair]: WebSocket;
   [kAccepted] = false;
   [kCoupled] = false;
   [kClosedOutgoing] = false;
   [kClosedIncoming] = false;
 
+  constructor(url: string | URL, protocols?: string | string[]);
+  constructor(flag: typeof kConstructOnly);
```

Comment:
I think this would be clearer (and more idiomatic) as a static method on WebSocket rather than a flag passed to the constructor, if that would be possible?

Reply:
Ah, I see, that makes sense—I didn't quite realise this was directly exposed